src/gg.regression.unity.bots/Runtime/Scripts/RGHeadlessSequenceRunner.cs

```diff
@@ -0,0 +1,255 @@
+﻿using System;
+using System.Collections;
+using System.Linq;
+using System.Reflection;
+using RegressionGames.StateRecorder.BotSegments;
+using RegressionGames.StateRecorder.BotSegments.Models;
+using RegressionGames.Types;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using Object = UnityEngine.Object;
+
+// ReSharper disable InconsistentNaming
+namespace RegressionGames
+{
+    public static class RGHeadlessSequenceRunner
+    {
+
+        internal static readonly int Rc_RunSuccess = 0;
+        internal static readonly int Rc_RunFailed = 1;
+        internal static readonly int Rc_SequencePathMissing = 2;
+        internal static readonly int Rc_OverlayNotInScene = 3;
+        internal static readonly int Rc_SequenceLoadFailure = 4;
+        internal static readonly int Rc_SequenceTimeoutNeedsPath = 5;
+        internal static readonly int Rc_SequenceTimeoutMissing = 6;
+        internal static readonly int Rc_SequenceTimeoutNotInt = 7;
+
+        internal static readonly string SequencePathArgument = "-rgsequencepath";
+        internal static readonly string SequenceTimeoutArgument = "-rgsequencetimeout";
+
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.BeforeSplashScreen)]
+        public static void FirstLoadChecks()
+        {
+            // do this here to fail fast on bad args
+            ParsePathArgument();
+            ParseTimeoutArgument();
+        }
+
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.AfterSceneLoad)]
+        public static void Initialize()
+        {
+            var resourcePath = ParsePathArgument();
+            if (resourcePath != null)
+            {
+                BotSegmentsPlaybackController playbackController = Object.FindObjectOfType<BotSegmentsPlaybackController>();
+                if (playbackController == null)
+                {
+                    RGDebug.LogError($"RGOverlay must be in the first scene loaded for your game to utilize {SequencePathArgument} command line argument");
```

Comment:
Do we see this in the console when running in headless mode? If not, I think we should definitely make a note in docs to point people to the correct log file to debug sudden closure of their app

Reply:
In console mode you get no STDOUT from Unity.  Everything is put into the -logfile path you specify, or defaults to the player.log file if not running in -batchmode

That said, I'm not very interested in re-documenting how Unity works for users 👎 

We absolutely should/will document that this feature exists once we've had a cycle or two of testing on it, 
https://linear.app/regression-games/issue/REG-2074/document-rgsequencepath-command-line-options